src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -260,3 +260,58 @@ function chol_blocked_rev(Σ̄::AbstractMatrix, L::AbstractMatrix, nb::Integer,
     # Convert to `Matrix`s because blas functions require StridedMatrix input.
     return chol_blocked_rev!(Matrix(Σ̄), Matrix(L), nb, upper)
 end
+
+#####
+##### `eigen`
+#####
+
+# k is the number of iterations of the power iteration that is 'used' in the forward pass,
+# this determines how many iterations are differentiated.
+# I am also iffy about the implementation of rrule for eigen in general, the paper
+# ignores the adjoints of the eigenvalues and also relies on the primal input being
+# symmetric positive semidefinite
+function rrule(::typeof(eigen), X::AbstractMatrix{<:Real};k=1)
+    # The paper only computes the adjoint using the adjoint of the eigenvectors,
+    # not the eigenvalues, and uses svd in the forward pass
+    F = eigen(X)
```

Comment:
Likewise, since the method only applies to positive semi-definite matrices, perhaps an error or warning should be thrown if any of the eigenvalues are negative?